cranelift-codegen/meta/src/cdsl/settings.rs

```diff
@@ -148,6 +148,14 @@ impl SettingGroup {
         }
         panic!("Should have found bool setting by name.");
     }
+
+    pub fn predicate_by_name(&self, name: &'static str) -> SettingPredicateNumber {
+        self.predicates
+            .iter()
+            .find(|pred| pred.name == name)
+            .expect(&format!("unknown predicate {}", name))
```

Comment:
nit: use `unwrap_or_else(|| panic!(…))` to avoid computing the string at every call.